changelog/2022-12-16T12_15_46+01_00_add_Clash_Magic_simulation

```diff
@@ -0,0 +1 @@
+ADDED: `Clash.Magic.simulation`, a way to differentiate between simulation and HDL translation.
```

Comment:
(My phrasing thingy. I forgot to check it into the review)

I just checked all occurrences of the word _HDL_ in `clash-prelude`. As I‍ suspected, we usually talk about generating HDL / generated HDL. (Also, I encountered "synthesized to HDL" which I object to.)

So I'd like to suggest to change "HDL translation / translating to HDL" to "HDL generation / generating HDL".

Reply:
RIP https://github.com/clash-lang/clash-compiler/issues/2279. But yeah, agreed.

Reply:
I'm fine with HDL being _generated_ for _translatable_ functions! So no need to bury #2279 yet. Translatability is a static property, generation a process. They can co-exist.

We'd just need to weed out that "synthesized to HDL" because I'd like to reserve the word synthesis and its companion word forms for what #2279 calls synthesisable code.